web-services/query/src/main/java/datawave/webservice/query/logic/filtered/QueryLogicFilterByParameter.java

```diff
@@ -0,0 +1,82 @@
+package datawave.webservice.query.logic.filtered;
+
+import datawave.webservice.query.Query;
+import datawave.webservice.query.QueryImpl;
+import org.apache.accumulo.core.security.Authorizations;
+
+import java.util.Set;
+
+public class QueryLogicFilterByParameter implements FilteredQueryLogic.QueryLogicFilter {
+    
+    // The parameter to match against
+    private String parameter;
+    
+    // The value to match against. If empty then boolean value of parameter is used.
+    private String value;
+    
+    // if negated than the negation of the match is returned
+    private boolean negated = false;
```

Comment:
Consider whether any of these can be `final`?

Reply:
No they cannot.  I need to be able to set them via setters to allow easier construction via spring (e.g. QueryLogicFactory.xml)